crates/router/src/core/payout_link.rs

```diff
@@ -248,40 +249,82 @@ pub async fn filter_payout_methods(
     > = HashMap::new();
-    let mut bank_transfer_hs: HashSet<common_enums::PaymentMethodType> = HashSet::new();
-    let mut card_hs: HashSet<common_enums::PaymentMethodType> = HashSet::new();
-    let mut wallet_hs: HashSet<common_enums::PaymentMethodType> = HashSet::new();
+    let mut bank_transfer_hash_set: HashSet<common_enums::PaymentMethodType> = HashSet::new();
+    let mut card_hash_set: HashSet<common_enums::PaymentMethodType> = HashSet::new();
+    let mut wallet_hash_set: HashSet<common_enums::PaymentMethodType> = HashSet::new();
+    let payout_filter_config = &state.conf.payout_filters.clone();
     for mca in &filtered_mca {
-        let payment_methods = match &mca.payment_methods_enabled {
+        let payout_methods = match &mca.payment_methods_enabled {
             Some(pm) => pm,
             None => continue,
         };
-        for payment_method in payment_methods.iter() {
+        for payout_method in payout_methods.iter() {
             let parse_result = serde_json::from_value::<api_models::admin::PaymentMethodsEnabled>(
-                payment_method.clone(),
+                payout_method.clone(),
             );
             if let Ok(payment_methods_enabled) = parse_result {
                 let payment_method = payment_methods_enabled.payment_method;
                 let payment_method_types = match payment_methods_enabled.payment_method_types {
-                    Some(pmt) => pmt,
+                    Some(payment_method_types) => payment_method_types,
                     None => continue,
                 };
-                for pmts in &payment_method_types {
-                    if payment_method == common_enums::PaymentMethod::Card {
-                        card_hs.insert(pmts.payment_method_type);
-                        payment_method_list_hm.insert(payment_method, card_hs.clone());
-                    } else if payment_method == common_enums::PaymentMethod::Wallet {
-                        wallet_hs.insert(pmts.payment_method_type);
-                        payment_method_list_hm.insert(payment_method, wallet_hs.clone());
-                    } else if payment_method == common_enums::PaymentMethod::BankTransfer {
-                        bank_transfer_hs.insert(pmts.payment_method_type);
-                        payment_method_list_hm.insert(payment_method, bank_transfer_hs.clone());
+                let connector = mca.connector_name.clone();
+                let payout_filter = payout_filter_config.0.get(&connector);
+                for request_payout_method_type in &payment_method_types {
+                    let currency_country_filter = check_currency_country_filters(
+                        payout_filter,
+                        request_payout_method_type,
+                        &payout.destination_currency,
+                        &address.country,
+                    )
+                    .await?;
+                    if currency_country_filter != Some(false) {
```

Comment:
```suggestion
                    if currency_country_filter == Some(true) {
```

Reply:
`None` here means Payoutmethod for that connector doesn't have configs in the application configs. For which we still want to show the payout method. `Some(false)` indicates that payout method filter is available but `currency`/ `country` is not present in the list. So we need to check for `Some(true)` or `None`.